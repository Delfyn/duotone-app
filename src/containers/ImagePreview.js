/* @flow */

import { connect } from 'react-redux';
import WallpaperPreview from './WallpaperPreview';

const mapStateToProps = state => ({
  primary: state.colors.primary,
  secondary: state.colors.secondary,
  fileName: state.files.fileName,
  originalFile: state.files.originalFile,
});

const WallpaperContainer = connect(mapStateToProps)(WallpaperPreview);

export default WallpaperContainer;
