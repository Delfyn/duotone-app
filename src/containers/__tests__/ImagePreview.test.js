import React from 'react';
import renderer from 'react-test-renderer';
import ImagePreview from '../ImagePreview';

test('ImagePreview renders correctly', () => {
  const state = {
    files: {
      originalFile: new Blob(),
      fileName: 'my-image.png',
    },
    colors: {
      primary: '#ff00ff',
      secondary: '#00ff00',
    },
  };

  const subscribe = jest.fn();
  const dispatch = jest.fn();
  const getState = jest.fn(() => state);

  const store = { subscribe, dispatch, getState };
  const props = {};

  const component = renderer.create(
    <ImagePreview store={store} {...props} />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  expect(subscribe).toBeCalled();
  expect(getState).toBeCalled();
});
