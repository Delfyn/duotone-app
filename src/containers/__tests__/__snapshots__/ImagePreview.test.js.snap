// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImagePreview renders correctly 1`] = `
<div
  className="image-background-wrapper flex items-center"
>
  <div
    className="image-background z1"
    style={
      Object {
        "backgroundImage": null,
      }
    }
  />
  <img
    alt="my-image.png"
    className="image-preview mx-auto z2"
    src=""
  />
</div>
`;
